test/emqx_protocol_SUITE.erl

```diff
@@ -19,114 +20,212 @@
-    io:format("~s", [emqx_packet:format(?PUBACK_PACKET(?PUBACK, 98))]),
-    io:format("~s", [emqx_packet:format(?PUBREL_PACKET(99))]),
-    io:format("~s", [emqx_packet:format(?SUBSCRIBE_PACKET(15, [{<<"topic">>, ?QOS0}, {<<"topic1">>, ?QOS1}]))]),
-    io:format("~s", [emqx_packet:format(?SUBACK_PACKET(40, [?QOS0, ?QOS1]))]),
-    io:format("~s", [emqx_packet:format(?UNSUBSCRIBE_PACKET(89, [<<"t">>, <<"t2">>]))]),
-    io:format("~s", [emqx_packet:format(?UNSUBACK_PACKET(90))]).
+-define(CLIENT2, ?CONNECT_PACKET(#mqtt_packet_connect{
+                                    username  = <<"admin">>,
+                                    clean_start = false,
+                                    password  = <<"public">>})).
 
-%%--------------------------------------------------------------------
-%% Message Cases
-%%--------------------------------------------------------------------
+%% receive_messages(Count) ->
+%%     receive_messages(Count, []).
 
-message_make(_) ->
-    Msg = emqx_message:make(<<"clientid">>, <<"topic">>, <<"payload">>),
-    ?assertEqual(0, Msg#message.qos),
-    Msg1 = emqx_message:make(<<"clientid">>, qos2, <<"topic">>, <<"payload">>),
-    ?assert(is_binary(Msg1#message.id)),
-    ?assertEqual(qos2, Msg1#message.qos).
-
-%% message_from_packet(_) ->
-%%     Msg = emqx_message:from_packet(?PUBLISH_PACKET(1, <<"topic">>, 10, <<"payload">>)),
-%%     ?assertEqual(1, Msg#message.qos),
-%%     %% ?assertEqual(10, Msg#message.pktid),
-%%     ?assertEqual(<<"topic">>, Msg#message.topic),
-%%     WillMsg = emqx_message:from_packet(#mqtt_packet_connect{will_flag  = true,
-%%                                                             will_topic = <<"WillTopic">>,
-%%                                                             will_payload   = <<"WillMsg">>}),
-%%     ?assertEqual(<<"WillTopic">>, WillMsg#message.topic),
-%%     ?assertEqual(<<"WillMsg">>, WillMsg#message.payload).
-
-    %% Msg2 = emqx_message:fomat_packet(<<"username">>, <<"clientid">>,
-    %%                                   ?PUBLISH_PACKET(1, <<"topic">>, 20, <<"payload">>)),
+%% receive_messages(0, Msgs) ->
+%%     Msgs;
+%% receive_messages(Count, Msgs) ->
+%%     receive
+%%         Other ->
+%%             ct:log("~p~n", [Other]),
+%%             receive_messages(Count-1, Msgs)
+%%     after 10 ->
+%%         Msgs
+%%     end.
 
+all() ->
+    [
+     {group, mqttv4},
+     {group, mqttv5}].
 
+groups() ->
+    [{mqttv4,
+      [sequence],
+      [
+       connect_v4,
+       subscribe_v4
+       %% unsubscribe_v4
+       %% publish_v4-
+      ]},
+     {mqttv5,
+      [sequence],
+      [
+       connect_v5,
+       subscribe_v5
+      ]
+     }].
+
+init_per_suite(Config) ->
+    emqx_ct_broker_helpers:run_setup_steps(),
+    Config.
+
+end_per_suite(_Config) ->
+    emqx_ct_broker_helpers:run_teardown_steps().
+
+connect_v4(_) ->
+    ct:print("MQTT v4 connect test starting"),
```

Comment:
I find all the ct prints unnecessary when using `rebar3 ct`, (`make rebar-ct` for this project).